app/services/liquid_assigns_service.rb

```diff
@@ -15,4 +18,44 @@ def assigns
 
   def assigns_pro
   end
+
+  # This method uses Liquid::VariableLookup to find all liquid variables from
+  # a given text. We use the list to know which project assign we need.
+  def assigns_from_content
+    return AVAILABLE_PROJECT_ASSIGNS if text.nil?
+
+    variable_lookup = Liquid::VariableLookup.parse(text)
+    return (variable_lookup.lookups & AVAILABLE_PROJECT_ASSIGNS)
```

Comment:
seeing this code and the intersect/validation, is L25 really needed? How does VariableLookup behave on empty input?

Reply:
See comment above. If the `:text` param is not passed, we need to fetch all the assigns.